### ISSUE TITLE
Ferium should only update mods which are 

### ISSUE THREAD

**Issue by theRookieCoder** [gorilla-devs/ferium#115]
## The problem
Currently Ferium 'deletes' mods that it doesn't recognise

## My solutions
Upgrading should be rewritten so that only mod files that are in Ferium's profile are updated. This is possible because of the new scan module which will be introduced

**Comment by KyleUltimate** [comment]
Hiw about Github Release?
~~I feel like github releases is giving more problems than it solves lol~~

**Comment by theRookieCoder** [comment]
Oh right it won't work. Maybe a seperate folder for GitHub releases mods that will be symlinked? I dunno

**Comment by theRookieCoder** [comment]
> ~I feel like github releases is giving more problems than it solves lol~

Yeah I guess there's a reason no one else did this, it's really sketchy.

**Comment by KyleUltimate** [comment]
Ehh, maybe slowly fade out the github release as an deprecated feature? Currently for upgrading we can use the way you proposed (but name it smth like github-mods instead). And since the scan feature is here! If ferium detects your github mod is also on modrinth/curseforge, give an option to migrate to it.
And in the far future, fade out this feature completely

**Comment by theRookieCoder** [comment]
Nah I want to keep it. In fact, I want to expand it further to include compiling from source.

**Comment by DarkShadow44** [comment]
Maybe store the name of the last mod that was downloaded, and only delete those? Or what exactly is the problem here?

**Comment by 64-Tesseract** [comment]
My idea: 2 folders, the main `mods` folder that Minecraft uses & a new `user-mods` folder. After running `upgrade`, Ferium will do what it needs to in `mods`, and then symlinks (or just copies) the contents of `user-mods` into `mods`.

...I could probably just write a bash script for this, but it would be good to have a per-profile "autorun" config that can run a script before or after an upgrade
